UndertaleModLib/Models/UndertaleGeneralInfo.cs

```diff
@@ -380,6 +432,8 @@ public void Serialize(UndertaleWriter writer)
     /// <inheritdoc />
     public void Unserialize(UndertaleReader reader)
     {
+        if (GMS2Version < GMSVersions.GMS2_3 && reader.GMS2_3)
+            GMS2Version = GMSVersions.GMS2_3;
```

Comment:
It's more a general safety check, as applied elsewhere. If the version has (somehow, I think in practice this is the first enum-affecting check to run) already been set to _greater_ than 2.3, we don't want to roll it back.

If you mean that it would never run because the reader should have already set it, it can't do that because the reader value is set at the start of reading but the GeneralInfo hasn't been created yet to store it. So it patiently bides its time until the GeneralInfo _is_ parsed, then sets it right at the start there.